packages/default-storage/example/examples/Functional.tsx

```diff
@@ -129,9 +130,9 @@ function Functional(): JSX.Element {
         }
       })
       .then(async () => {
-        const AsyncStorageTestSupport = TurboModuleRegistry.get(
-          "AsyncStorageTestSupport"
-        );
+        const AsyncStorageTestSupport = TurboModuleRegistry
+          ? TurboModuleRegistry.get("AsyncStorageTestSupport")
+          : NativeModules["AsyncStorageTestSupport"];
```

Comment:
TurboModuleRegistry already fallsback to NativeModules, so no need to check both.

Reply:
When calling TurboModuleRegistry directly, the `tsc` check was throwing the following error: https://github.com/react-native-async-storage/async-storage/actions/runs/11295316739/job/31491051486#step:5:26. I tried changing the call to `TurboModuleRegistry.get<TurboModule>` in case it was a typing issue, but same result.

Reply:
You're getting typing issues because `TurboModule` is a generic type, it doesn't contain the two methods that are being called further down. You need to first define the types:

```typescript
type AsyncStorageDelegate = () => void;

type AsyncStorageTestSupport = {
  test_setDelegate: (delegate: AsyncStorageDelegate) => void;
  test_unsetDelegate: (delegate: AsyncStorageDelegate) => void;
};
```

Then use it like this:

```typescript
const AsyncStorageTestSupport = TurboModuleRegistry.get<AsyncStorageTestSupport>("AsyncStorageTestSupport");
```